dash/testing/browser.py

```diff
@@ -247,16 +247,22 @@ def open_new_tab(self, url=None):
         )
 
     def get_webdriver(self, remote):
-        return (
-            getattr(self, "_get_{}".format(self._browser))()
-            if remote is None
-            else webdriver.Remote(
-                command_executor=remote,
-                desired_capabilities=getattr(
-                    DesiredCapabilities, self._browser.upper()
-                ),
-            )
-        )
+        # occasionally the browser fails to start - give it 3 tries
+        for i in reversed(range(3)):
+            try:
+                return (
+                    getattr(self, "_get_{}".format(self._browser))()
+                    if remote is None
+                    else webdriver.Remote(
+                        command_executor=remote,
+                        desired_capabilities=getattr(
+                            DesiredCapabilities, self._browser.upper()
+                        ),
+                    )
+                )
+            except WebDriverException:
+                if not i:
+                    raise
```

Comment:
@byronz I had 3 failures this morning - all from different tests, but with the same message:
```
WebDriverException: Message: unknown error:
Chrome failed to start: exited abnormally
(unknown error: DevToolsActivePort file doesn't exist)
```

It's a bit icky to add a blind retry loop like this, but it seemed to work.

Reply:
I had an impression this crash comes always from one specific python version, like 3.7, or 2.7.   Might be good to figure out the real root cause, but assuming it's just turbulence from the circleci docker. it's more related to bad timing due to load performance of the selenium server.   should we try another approach like the `wait.until`? with a reasonable small wait time like 0.1s, I believe the next try should work (we can give a more tolerant 1s timeout).

Reply:
@alexcjohnson  I created another PR to fix the deprecated message with `pytest.raises`, and I formatted the file with `black`. so if you can format this test_integration file in this PR, we could avoid a lot potential merge conflicts.

Reply:
I saw this in both 2.7 and 3.7. Feel free to dig in more (preferably in another PR so we can get this bugfix merged), but it's not clear to me what we could wait for, seems like this is happening deep inside Selenium.

Reply:
agree, we can leave it now in this PR.